demo/rust/filedemo/src/main.rs

```diff
@@ -0,0 +1,153 @@
+/*
+    Copyright 2021 Picovoice Inc.
+
+    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+    file accompanying this source.
+
+    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+    specific language governing permissions and limitations under the License.
+*/
+
+use chrono::Duration;
+use clap::{App, Arg, ArgGroup};
+use itertools::Itertools;
+use rhino::{BuiltinContexts, RhinoBuilder};
+use rodio::{source::Source, Decoder};
+use std::fs::File;
+use std::io::BufReader;
+use std::path::PathBuf;
+
+fn rhino_demo(
+    input_audio_path: PathBuf,
+    context_path: Option<&str>,
+    context_builtin: Option<BuiltinContexts>,
+    sensitivity: Option<f32>,
+    model_path: Option<&str>,
+) {
+    let soundfile = BufReader::new(File::open(input_audio_path).unwrap());
+    let audiosource = Decoder::new(soundfile).unwrap();
+
+    let mut rhino_builder = match context_path {
+        Some(context_path) => RhinoBuilder::new(context_path),
+        None => RhinoBuilder::new_with_builtin(context_builtin.unwrap()),
+    };
+
+    if let Some(sensitivity) = sensitivity {
+        rhino_builder.sensitivity(sensitivity);
+    }
+
+    if let Some(model_path) = model_path {
+        rhino_builder.model_path(model_path);
+    }
+
+    let rhino = rhino_builder.init().expect("Failed to create Rhino");
+
+    if rhino.sample_rate() != audiosource.sample_rate() {
+        panic!(
+            "Audio file should have the expected sample rate of {}, got {}",
+            rhino.sample_rate(),
+            audiosource.sample_rate()
+        );
+    }
+
+    let mut timestamp = Duration::zero();
+    for frame in &audiosource.chunks(rhino.frame_length() as usize) {
+        let frame = frame.collect_vec();
+        timestamp = timestamp
+            + Duration::milliseconds(((1000 * frame.len()) / rhino.sample_rate() as usize) as i64);
+
+        if frame.len() == rhino.frame_length() as usize {
+            let is_finalized = rhino.process(&frame).unwrap();
+            if is_finalized {
+                let inference = rhino.get_inference().unwrap();
+                if inference.is_understood {
+                    println!(
+                        "[{}:{}:{}] Detected {{",
```

Comment:
maybe kick the json print down to a new line so that the structure is nice and clear